application/ui/components/input/array.jsx

```diff
@@ -0,0 +1,147 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'ArrayParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values   : [],
+            numItems : 0
+        };
+    },
+
+    getValue : function()
+    {
+        var data = [];
+        for(var i in this.state.values) {
+            data.push(this.state.values[i]);
+        }
+        return data;
+    },
+
+    getInputs : function()
+    {
+        var inputs    = [],
+            component = this;
+
+        _.each(this.state.values, function (value, itemNum) {
+            inputs.push(component.getInput(value, itemNum));
+        });
+
+        return inputs;
+    },
+
+    getInput : function(value, itemNum)
+    {
+        var instance = this;
+        var ParamRow = require('../param-row');
+
+        var rmCallback = function () {
+            instance.removeField(itemNum);
+        };
+
+        return (
+            <tr key={itemNum}>
+                <td>
+                    <a className='button field-button--remove' onClick={rmCallback}>–</a>
+                </td>
+                <td>
+                    <table>
+                    <ParamRow
+                        simple      = {true}
+                        value       = {value}
+                        key         = {itemNum}
+                        param       = {instance.props.param}
+                        onChange    = {_.partial(instance.changeHandler, itemNum)}
+                        onInclude   = {function() {}}
+                        onNull      = {function() {}}
```

Comment:
We can avoid having to supplying empty functions by returning them in the `getDefaultProps` method of the `ParamRow` component.